servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -289,12 +283,16 @@ private HttpExecutionContext buildExecutionContext(final HttpExecutionStrategy s
      * </dl>
      *
      * @param rawService {@link StreamingHttpService} to use for the server.
-     * @param serviceStrategy the {@link HttpExecutionStrategy} to use for the service.
+     * @param apiStrategy The {@link HttpExecutionStrategy} provided by the user/API
      * @return A {@link Single} that completes when the server is successfully started or terminates with an error if
      * the server could not be started.
      */
     private Single<HttpServerContext> listenForService(final StreamingHttpService rawService,
-                                                       final HttpExecutionStrategy serviceStrategy) {
+                                                       final HttpExecutionStrategy apiStrategy) {
+        HttpExecutionStrategy serviceStrategy = computeServiceStrategy(rawService);
+        HttpExecutionStrategy serverStrategy =
+                apiStrategy.hasOffloads() ? apiStrategy.merge(serviceStrategy) : apiStrategy;
```

Comment:
1. Consider adding an assertion that `apiStrategy` is never `defaultStrategy()`
2. When `apiStrategy.hasOffloads() == false`, why do we ignore the result of `computeServiceStrategy`?

Reply:
The behaviour of the builder `executionStrategy()` is effectively to override the filter and service strategy. In this case the builder strategy was likely either `offloadNone()` or `offloadNever()` indicating that the user desired no offloading.